aws/credentials/credentials.go

```diff
@@ -21,18 +22,20 @@ type awsCred struct {
 
 func NewAWSCredentials(id, secret, token string, runningOnEc2 bool) (AWSCredentials, error) {
 	var creds *credentials.Credentials
+	sess := session.Must(session.NewSession())
```

Comment:
A quick note, don't we want to use this instead ::

```go
sess, err := session.NewSession()
if err != nil { ...
```

As far as I can tell --
* AWS's `session.NewSession()` is not a trivial func -- there's plenty of opportunity for "things to go bad', and  `session.Must()` will throw PANIC when that happens, also
* the `NewAWSCredentials()` is allowed to return `error` -- so it becomes trivial to escalate `session.NewSession()` error back to the caller..